cvmfs/catalog_mgr_impl.h

```diff
@@ -879,23 +905,34 @@ CatalogT *AbstractCatalogManager<CatalogT>::MountCatalog(
                                               CatalogT *parent_catalog)
 {
   CatalogT *attached_catalog = NULL;
-  if (IsAttached(mountpoint, &attached_catalog))
+  if (IsAttached(mountpoint, &attached_catalog)) {
     return attached_catalog;
+  }
+
+  CatalogContext ctlg_context(hash, mountpoint, kCtlgLocationMounted);
```

Comment:
i will answer here also for your change request below:

`LoadFreeCatalog` is only called for nested mountpoints. the catalog location is only important for the root catalog, as this means that you might want to cache the manifest locally. so here we dont care.

`MountCatalog` can be called both for nested and root catalogs. and the root catalog can either be a fixed hash or null. the location of the root catalog (fixed or not) can be local or remote.

there is now 1 border case i did not see before: if it is a fixed root catalog with the current setting `kCtlgLocationMounted` it will not safe the manifest.
problem is: we also do not have the manifest. and manifest::Fetch should return the newest manifest, not the manifest related to the hash?

in anyway i guess we should add the feature to verify the manifest for fixed root catalogs?

Reply:
added `kCtlgNoLocationNeeded`